### PR TITLE
Allow Drupal to wrap the Symfony test listener (5.1 backport)

### PR DESCRIPTION
**Pull request by alexpott**
| Q             | A
| ------------- | ---
| Branch?       | 5.1
| Bug fix?      | kinda
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #... 
| License       | MIT
| Doc PR        | symfony/symfony-docs#... 

This is a backport of #37708

### CODE REVIEW

File: src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler/Deprecation.php [hunk 1]
```diff
@@ -61,10 +60,10 @@ public function __construct($message, array $trace, $file)
         $line = $trace[$i];
         $this->triggeringFile = $file;
         if (isset($line['object']) || isset($line['class'])) {
-            if (isset($line['class']) && 0 === strpos($line['class'], SymfonyTestsListenerFor::class)) {
-                set_error_handler(function () {});
-                $parsedMsg = unserialize($this->message);
-                restore_error_handler();
+            set_error_handler(function () {});
+            $parsedMsg = @unserialize($this->message);
```
Comment (nicolas-grekas) [inline src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler/Deprecation.php:64]: the `@ is not required anymore since the error handler ignores everything

Comment (alexpott) [reply src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler/Deprecation.php:64]: It's in master - that's why I added it back.

Comment (alexpott) [reply src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler/Deprecation.php:64]: I'll remove it here and submit a PR to clean up master.

Comment (alexpott) [reply src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler/Deprecation.php]: Created #38033 to address this in master.

**Review by nicolas-grekas — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @alexpott.

**Pull request merged**